internal/ir/xds.go

```diff
@@ -319,6 +324,29 @@ type JWT struct {
 	Providers []egv1a1.JWTProvider `json:"providers,omitempty" yaml:"providers,omitempty"`
 }
 
+// OIDC defines the schema for authenticating HTTP requests using
+// OpenID Connect (OIDC).
+//
+// +k8s:deepcopy-gen=true
+type OIDC struct {
+	// The OIDC Provider configuration.
+	Provider egv1a1.OIDCProvider `json:"provider"`
+
+	// The OIDC client ID assigned to the filter to be used in the
+	// [Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest).
+	ClientID string `json:"clientID"`
+
+	// The Kubernetes secret which contains the OIDC client secret assigned to the filter to be used in the
+	// [Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest).
+	//
+	// This is an Opaque secret. The client secret should be stored in the key "clientSecret".
+	ClientSecret gwapiv1b1.SecretObjectReference `json:"clientSecret"`
```

Comment:
this should be []byte similar to https://github.com/envoyproxy/gateway/blob/c85f0e438ec3c41048739e794f27f3a33e853df0/internal/ir/xds.go#L219

https://github.com/envoyproxy/gateway/blob/c85f0e438ec3c41048739e794f27f3a33e853df0/internal/ir/xds.go#L147 also will need to get updated

Reply:
The OIDC client secret will not be used as an inline string like TLS private key does, instead, it will be fetched from the EG xDS server through SDS, so we don't need []byte in the IR.